forem/forem
# Ensure Article's `username` method returns a downcased string

## What type of PR is this? (check all applicable)


## Description

~A user's posts were inaccessible because the `path` in their articles didn't match the query with the downcase params. This PR updates the way the query is done in Postgres by lowercasing the fields.~

A new approach: Instead of updating the logic in `stories#show` this PR ensures the path is built forcing to lowercase `calculated_path` in `set_caches` (`before_save` callback)

Check out this query (capitalized username): https://dev.to/admin/blazer/queries/303-issue-11690

## Related Tickets & Documents

Closes #11690 

## QA Instructions, Screenshots, Recordings

1. Make your own user have a capitalized username
1. Publish a new Article
1. Check out your newly created Article
1. Edit the same Article and save

None of the above should break the app

### UI accessibility concerns?

N/A

## Added tests?


## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## Post Deployment tasks

Make sure these links work after the data update script runs:
- http://dev.to/AlbertoFdzM/mongoose-mongodb-and-express-596
- https://dev.to/maxime1992/manage-your-dev-to-blog-posts-from-a-git-repo-and-use-continuous-deployment-to-auto-publish-update-them-143j
- https://dev.to/TomGranot/quick-tip-increase-keyboard-repeat-rate-for-faster-typing-4d8g

Also check that this Blazer query returns 0 rows:
https://dev.to/admin/blazer/queries/304-articles-with-uppercased-characters

## [optional] What gif best describes this PR or how it makes you feel?

![sort of the same thing](https://media.giphy.com/media/xBIuPT5pjvarJZXyBw/giphy.gif)
